packages/idempotency/src/PersistenceLayer.ts

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { IdempotencyConfig } from './IdempotencyConfig';
+import { PersistenceLayerInterface } from './PersistenceLayerInterface';
+
+abstract class PersistenceLayer implements PersistenceLayerInterface {
+  public configure(_config: IdempotencyConfig): void {}
+  public deleteRecord(): void {}
+  public getRecord(): void {}
+  public saveInProgress(): void {}
+  public saveSuccess(): void {}
+
+  protected abstract _deleteRecord(): void;
+  protected abstract _getRecord(): void;
+  protected abstract _putRecord(): void;
+  protected abstract _updateRecord(): void;
+}
+
+class DynamoDBPersistenceLayer extends PersistenceLayer {
```

Comment:
Should we have a separate file/location for the dynamodb persistance layer?  I remember we discussed it at some point.

Reply:
yeah I think that would be a good idea. Python also has these in a "persistence" sub folder. We can follow some of that file structure - if not we can discuss it

Reply:
I moved this into a persistence folder